application/models/Project_model.php

```diff
@@ -363,6 +363,80 @@ function get_frozen_projects()
         return $query->result();
 
     }
+    
+    
+    public function get_projects_by_keywords_id($params)
+    {
+	// If calls to this function do not pass a project_type_description parameter,
+	// provide a sensible default
+	
+	// It is likely that either $params['project_type_description'] will not exist as a key at all,
+	// or that if it does exist, it will contain the values of 'either','solo' or 'group', matching values passed
+	// from Javascript embedded in the search page footer. Of these values, only 'solo' matches a value actually to be
+	// found in the projects.project_type field.
+	if ((!array_key_exists('project_type_description', $params)) ||  ($params['project_type_description'] == 'either'))
+	{
+		$params['project_type_description'] = "%" ; 
+	}
+	if ($params['project_type_description'] == 'group')
+	{
+		$params['project_type_description'] = 'collaborative'; 
+	}
+	
+        $offset = 0 + $params['offset'];
+        $limit = 0 + $params['limit'];      
+        
+        $sql = 'select p.id, p.title_prefix, p.title, COALESCE(p.date_catalog, "2001-01-01" ) as safe_release_date, 
+        p.url_librivox, p.status, p.coverart_thumbnail, 	        p.zip_url, p.zip_size, l.two_letter_code, l.language, p.url_forum, p.project_type
+	FROM keywords k
+	JOIN project_keywords pk ON (k.id = pk.keyword_id)
+	JOIN projects p on (pk.project_id = p.id)
+	JOIN languages l ON (l.id = p.language_id) ';
+	$sql .= 'WHERE k.id = ' . $params['keywords_id'] . ' ';	
```

Comment:
I think this is probably safe because nothing in `$params` actually comes from user input (see two commits after this one, where this method is being called), but I always get very nervous when we do string concatenation of variables into SQL statements. It might be safe _now_, but to ascertain that we have to check every instance of this being called. We have enough SQL injection as is, could we rewrite this as an entirely parametrized query?